### DISCUSSION TITLE
AWS S3 type hints

### DISCUSSION THREAD

**Discussion by rozkomnadzor** [dagster-io/dagster#16551] [Q&A]
There is a way to enable type hints for s3 client?
I use `s3: S3Resource`, `client = s3.get_client()`, but any classes (such as BaseClient from botocore) do not give any hints of methods that can be used

**Comment by manelfideles** [comment]
Any updates on this? :(

**Reply by garethbrickman** [reply]
I haven't tried this out, but our LLM suggested this: 

You can manually add type hints to your code to improve the development experience.

```python
from typing import TYPE_CHECKING
from dagster_aws.s3 import S3Resource

if TYPE_CHECKING:
    from botocore.client import BaseClient

def example_s3_op(s3: S3Resource):
    client: 'BaseClient' = s3.get_client()
    response = client.list_objects_v2(Bucket='my-bucket', Prefix='some-key')
    return response
```

In this example, the TYPE_CHECKING import ensures that the type hint for BaseClient is only evaluated during type checking and not at runtime. This allows your IDE or type checker to provide method hints for the client object.

**Reply by arnimarj** [reply]
As a suggestion, I've used https://youtype.github.io/boto3_stubs_docs/ extensively, and I would recommend it to every project using `boto3`.
